docs-chef-io/content/server/auth.md

```diff
@@ -386,6 +386,61 @@ common causes:
     minutes. This can be fixed by syncing the clock with an Network Time
     Protocol (NTP) server.
 
+## User public key update
+
+Chef infra server user can update public_key using the following methods.
+-   Using reregister method
+    Use the reregister argument in the knife command to regenerate an RSA key pair for a user. The public key will be stored on the Chef Infra Server and the private key will be displayed on STDOUT or written to a named file.
+    
+    Syntax:- knife user reregister USER_NAME (options)
+    Ex:- knife user reregister "Arjun Koch" 
+
+-  Using knife user key command
+    User can have multiple public keys associated with his chef server account.  Each public key has a name and expiration date associated with it.
+    Name is an ASCII string used to identify the public key.  The expiration date for the public key is specified as an ISO 8601 formatted string: YYYY-MM-DDTHH:MM:SSZ.
+    User can list, add, edit and delete public keys with knife user key command.  The following operations supported.
+        knife user key create to create public key.
+        knife user key delete to delete public key.
+        knife user key list to list user public keys.
+        knife user key edit to edit public keys.
+
+    Constraints: User can't edit key details which is currently in use.
```

Comment:
Does this just mean that if I'm using the `user1` key, I can't regenerate a new `user1` key? I have to create or use another key first and then regenerate user1 while using the other key?

Reply:
Does this also apply to the `knife user reregister` command?

Reply:
I mean user can't edit details of the key which he is using to send current request.

**Does this also apply to the knife user reregister command?**
    No reregister command will always edit public key named default, create if one dose not exists.
    if current request is made with help of default private key then also request will succeed.